crates/router/src/core/payments/helpers.rs

```diff
@@ -2028,7 +2030,7 @@ pub async fn make_pm_data<'a, F: Clone, R>(
     }
 
     // TODO: Handle case where payment method and token both are present in request properly.
-    let (payment_method, pm_id) = match (request, payment_data.token_data.as_ref()) {
```

Comment:
I don't think this clone is required? u can match on reference of `request` itself